### ISSUE TITLE
Webaudio default sink missing

### ISSUE THREAD

**Issue by clinique** [openhab/openhab-core#821]
I got this error message since few days :  
`Default AudioSink service 'webaudio' not available!`

**Comment by wborn** [comment]
The Webaudio sink is now part of the openhab2-addons repo and has a feature but it's not installed by any other default installed feature. It's also not installable using the UIs. I tried installing it with `feature:install openhab-io-webaudio` but it still wasn't added to the list of audio sinks.

It used to be part of the openhab-runtime-base feature but that is in the openhab-core repo, so features part of openhab2-addons cannot be added to it. 

I see openhab-io-javasound is installed by the [openhab-misc-ruleengine feature](https://github.com/openhab/openhab2-addons/blob/3b47874a0fa503f699a1edbc80c38389c5848876/features/openhab-addons/src/main/resources/footer.xml#L28). Do you think the webaudio bundle should be installed similarly or by the standard/demo/expert packages in openhab-distro @kaikreuzer?

**Comment by maggu2810** [comment]
> It used to be part of the openhab-runtime-base feature but that is in the openhab-core repo, so features part of openhab2-addons cannot be added to it.

IMHO `openhab-runtime-base` should be moved to openHAB distro and no other feature should depend on it. It is just the collection of bundles that is installed default by the distribution. Or am I wrong?

**Comment by kaikreuzer** [comment]
> Do you think the webaudio bundle should be installed similarly or by the standard/demo/expert packages in openhab-distro @kaikreuzer?

How about adding it to the web UIs, which actually make use of it (Paper UI, HABPanel and HABot afaik)? If no UI is installed, it probably isn't needed either.

> I tried installing it with feature:install openhab-io-webaudio but it still wasn't added to the list of audio sinks.

This is weird, I would have expected that this works.

> IMHO openhab-runtime-base should be moved to openHAB distro and no other feature should depend on it.

Currently, ALL other features depend on it. How should it look like without that?

**Comment by kaikreuzer** [comment]
> How about adding it to the web UIs, which actually make use of it

I just checked, this is actually already the case.

The issue is that the feature does not contain any bundle:
```
openhab> feature:info openhab-io-webaudio 
Feature openhab-io-webaudio 2.5.0.SNAPSHOT
Description:
  Web Audio Support
Feature has no configuration
Feature has no configuration files
Feature depends on:
  openhab-runtime-base 0.0.0
Feature has no bundles.
Feature has no conditionals.
```
Indeed the packaged feature.xml does not list it:
```
    <feature name="openhab-io-webaudio" description="Web Audio Support" version="2.5.0.SNAPSHOT">
        <feature>openhab-runtime-base</feature>
    </feature>
```
The [source fragment](https://github.com/openhab/openhab2-addons/blob/master/bundles/org.openhab.io.webaudio/src/main/feature/feature.xml) looks ok to me, though. How exactly is the addons feature.xml now assembled? What goes wrong there?

**Comment by wborn** [comment]
> How about adding it to the web UIs, which actually make use of it (Paper UI, HABPanel and HABot afaik)? 

That makes a lot of sense indeed.

> What goes wrong there?

Seems the regexp wasn't matching some bundles but it should be fixed with https://github.com/openhab/openhab2-addons/pull/5731. I diffed the differences and the openhab-misc-market feature was also missing its bundle.

**Comment by maggu2810** [comment]
> Currently, ALL other features depend on it. How should it look like without that?

Not really all, AFAIK no core feature depends on it.
I don't know why every addon (also web ui etc.) needs to depend on that.
IMHO a feature should depend on what it needs and not on a bunch of other not needed stuff.
As `openhab-runtime-base` is already installes by the distribution, there is no need to hardly depend on it.
Using `openhab-runtime-base` in every feature has the "advantage" that all the feature repositories (openhab2-addons, webui, ...) are rather useless for non official openHAB distributions.
As soon as an feature is installed a bunch of bundles are installed that you perhaps don't want to provide / use / support.

**Comment by kaikreuzer** [comment]
> That makes a lot of sense indeed. I see Paper UI already depends on openhab-io-webaudio whereas HABPanel and HABot don't.

I've created https://github.com/openhab/openhab-webui/pull/81 for it.

> Seems the regexp wasn't matching some bundles but it should be fixed with openhab/openhab2-addons#5731.

Many thanks! I can confirm that the Web Audio option is now available when Paper UI is installed in build 1614, so this issue is fixed.

> Not really all, AFAIK no core feature depends on it.

Right, but it is the "base" feature for the add-ons. I think this was already discussed in the past. It makes a lot of things easier for contributors, if they don't have to learn about any fine-grained split of features, but if we provide the right set through a single feature. I agree that it isn't nice for "outside openHAB use" and if this is something that we want to simplify and nicely support, it can be discussed. A good compromise might be a feature per "extension type", i.e. a feature that provides everything relevant for bindings, etc. But this is totally unrelated to this issue here, so let's move the dicussion, if you want to follow-up on it.
